From: Andrew Lunn <andrew@lunn.ch>
Date: Mon, 20 Apr 2020 00:11:52 +0200
Subject: net: ethtool: self_test: Mark interface in testing operative status
Patch-mainline: v5.8-rc1
Git-commit: 77e9b2ab451d32c81468ef679c377b2f831cd720
References: bsc#1176447

When an interface is executing a self test, put the interface into
operative status testing.

Signed-off-by: Andrew Lunn <andrew@lunn.ch>
Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 net/ethtool/ioctl.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/net/ethtool/ioctl.c
+++ b/net/ethtool/ioctl.c
@@ -1747,7 +1747,9 @@ static int ethtool_self_test(struct net_
 	if (!data)
 		return -ENOMEM;
 
+	netif_testing_on(dev);
 	ops->self_test(dev, &test, data);
+	netif_testing_off(dev);
 
 	ret = -EFAULT;
 	if (copy_to_user(useraddr, &test, sizeof(test)))
